### DISCUSSION TITLE
文言文程式語言

### DISCUSSION THREAD

**Discussion by jerry850313** [1101-GS4719-C/Python-Community-GS4719-C#94] [Ideas]
雖然這個有點舊，但還是分享一下，畢竟這個[東西](https://buzzorange.com/techorange/2019/12/18/coding-in-ancient-chinese/)很有趣。
![image](https://buzzorange.com/techorange/wp-content/uploads/sites/2/2019/12/EL-k-zDW4AADGKD.jpg)

**Comment by xCk27x** [comment]
程式語言中的變數跟關鍵字很好的被替換成中文字了，如 if 變成了「若」，for 變成了「凡」。但如果要用文言文來表達位元運算的話，感覺一定會超奇怪，像左移5位元之類的，我完全想不出來怎麼用文言文表達。

**Reply by jerry850313** [reply]
可以去他[github](https://github.com/wenyan-lang/wenyan)看看，也許有解決方法

**Comment by 109802503gloria** [comment]
不只是指令全變成中文字了，連標點符號都被替換了，雖然文言文已經是精簡的中文，但還是無法達到英文的便利性，只能說他看起來很酷。
至於文章中說可以同時學習程式與文言文我認為是不可行的，畢竟程式是有固定格式的，在裡面除了變數有就只能看到關鍵字，太侷限了。

**Reply by jerry850313** [reply]
所以才說有趣

**Comment by Jen-Siang** [comment]
文言文閱讀上有困難，英文理解還是比較快，加上全球多以英文為溝通語言，若使用文言文可能會有諸多不便。但回歸貼文，這程式的確很有趣

**Reply by wax75397** [reply]
推個，文言文真的比較難讀，有些華人都看不懂了，外國人就更不用說了

**Comment by chun1203** [comment]
相當的有趣，感覺像在講故事XD

**Comment by JAIMEHUNG** [comment]
好酷!我覺得比較好懂欸(我一類組)
教育部要不要考慮加進國文考題XD

**Comment by Jie2222** [comment]
很有趣的想法，因為不習慣所以會覺得看不太懂，但也許將來也會開發中文程式語言

**Comment by zixunn** [comment]
非常有趣但讓人看得好累

**Comment by ghost** [comment]
哈哈好酷的東西 全世界最人的母語是中文，說不定之後程式碼......XD

**Comment by cody910120** [comment]


**Reply by cody910120** [reply]
> 文言文-->很難 程式語言-->很難 文言文+程式語言-->難上加難

**Comment by yt7l0** [comment]
這兩種語言果然不能混用啊……

**Comment by YuHengChen54** [comment]
普通文言文就看不懂了...還拿來打程式，跟之前那個清代的微積分課本有得拚

**Comment by chupochun** [comment]
也太帥了吧，文言文都看不懂了，還拿來當成程式語言。

**Reply by pekoranoinu** [reply]
呵

**Comment by pekoranoinu** [comment]
原本就看不懂的東西，把它變文言文更看不懂了...

**Comment by bloodtrail0511** [comment]
右邊還能從文字裡面猜到程式碼在幹嘛，結果一換成中文連保留字都看不懂了哈哈

**Comment by ThomasMeng0515** [comment]
這樣識別字跟函數區塊真的好難辨別，發明這個文言文程式的也太有才

**Comment by ghost** [comment]
雖然有趣但感覺蠻沒意義的語言，不過能研發出好東西都值得肯定

**Comment by d26488399** [comment]
正常的文言文都看不懂了，變程式碼更可怕了，但是如果學這種程式語言，好像反而可以讓國文變好。

**Comment by Sazmafuji** [comment]
如果要這樣寫code，我認為我會暴斃

**Comment by sean1325418** [comment]
研發這種程式碼到底是搞別人還是搞自己呢

**Comment by yi-1111** [comment]
太神了，沒想到還可以這樣玩

**Comment by Yuhsin000** [comment]
還是英文舒服些，以後會珍惜看英文程式的難能可貴
